src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/bfaction/BfActionReportTest.java

```diff
@@ -0,0 +1,282 @@
+        var caseData = getTestCaseData();
+        List<BFActionTypeItem> items = getBFActionTypeItemsWithClearedBfAction();
+        var bfActionTypeItem2 = new BFActionTypeItem();
+        bfActionTypeItem2.setId("456");
+        var bFActionType2 = new BFActionType();
+        bFActionType2.setCwActions("Interlocutory order requested");
+        bFActionType2.setBfDate("2019-12-14T13:30:55");
+        bFActionType2.setDateEntered("2019-11-20");
+        bFActionType2.setCleared("2019-12-15");
+        bFActionType2.setNotes("test non-cleared bf");
+        bfActionTypeItem2.setValue(bFActionType2);
+        items.add(bfActionTypeItem2);
+        caseData.setBfActions(items);
+
+        var bfActionReport = new BfActionReport();
+        var submitEvent = new SubmitEvent();
+        submitEvent.setCaseId(2);
+        submitEvent.setState(ACCEPTED_STATE);
+        submitEvent.setCaseData(caseData);
+        submitEvents.add(submitEvent);
+
+        var resultListingData = bfActionReport.runReport(listingDetails, submitEvents);
+        var expectedBfDateCount = 2;
+        assertEquals(expectedBfDateCount, resultListingData.getBfDateCollection().size());
+    }
+
+    @Test
+    public void shouldNotReturnClearedBfActionsWithInDateRange() {
+        var caseData = getTestCaseData();
+        List<BFActionTypeItem> items = getBFActionTypeItemsWithClearedBfAction();
+        caseData.setBfActions(items);
+        var bfActionReport = new BfActionReport();
+        var submitEvent = new SubmitEvent();
+        submitEvent.setCaseId(2);
+        submitEvent.setState(ACCEPTED_STATE);
+        submitEvent.setCaseData(caseData);
+        submitEvents.add(submitEvent);
+
+        var resultListingData = bfActionReport.runReport(listingDetails, submitEvents);
+        var expectedBfDateCount = 2;
+        assertEquals(expectedBfDateCount, resultListingData.getBfDateCollection().size());
+    }
+
+    @Test
+    public void shouldReturnBfActionsSortedByBfDate() {
+        listingData.setListingDateFrom("2019-12-08");
+        listingData.setListingDateTo("2019-12-25");
+        listingData.setHearingDateType(RANGE_HEARING_DATE_TYPE);
+        listingDetails.setCaseData(listingData);
+        var caseData = getTestCaseData();
+        List<BFActionTypeItem> items = getBFActionTypeItemsWithClearedBfAction();
+
+        var bfActionTypeItem6 = new BFActionTypeItem();
+        bfActionTypeItem6.setId("456");
+        var bFActionType6 = new BFActionType();
+        bFActionType6.setCwActions("Interlocutory new order requested");
+        bFActionType6.setBfDate("2019-12-08");
+        bFActionType6.setDateEntered("2019-11-20");
+        bFActionType6.setNotes("test non-cleared bf sixth");
+        bfActionTypeItem6.setValue(bFActionType6);
+        items.add(bfActionTypeItem6);
+        caseData.setBfActions(items);
+        var bfActionReport = new BfActionReport();
+
+        var submitEvent = new SubmitEvent();
```

Comment:
Maybe put submitEvent in the setUp as it's being used in multiple tests as opposed to declaring it each time

Reply:
same with bfActionReport

Reply:
good point. Same with BfActionReport in the tests.